website/blog/2024-06-20-run-ferretdb-cloudnativepg-kubernetes.md

```diff
@@ -0,0 +1,370 @@
+          image: ghcr.io/ferretdb/ferretdb:latest
+          ports:
+            - containerPort: 27017
+          env:
+            - name: FERRETDB_POSTGRESQL_URL
+              value: 'postgresql://ferretuser@postgres-cluster-rw.cnpg.svc.cluster.local:5432/ferretdb'
+
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: ferretdb
+  namespace: cnpg
+spec:
+  selector:
+    app: ferretdb
+  ports:
+    - protocol: TCP
+      port: 27017
+      targetPort: 27017
+  type: NodePort
+```
+
+Apply the `yaml` config:
+
+```sh
+kubectl apply -f ferretdeployment.yaml
+```
+
+Confirm that all essentials resources have been created:
+
+```sh
+kubectl get cluster,pod,svc,secret -n cnpg
+```
+
+Output:
+
+```text
+NAME                                          AGE   INSTANCES   READY   STATUS                     PRIMARY
+cluster.postgresql.cnpg.io/postgres-cluster   17m   3           3       Cluster in healthy state   postgres-cluster-1
+
+NAME                                       READY   STATUS    RESTARTS   AGE
+pod/cnpg-cloudnative-pg-6685f6c7b8-5w8pp   1/1     Running   0          20m
+pod/ferretdb-f6864c9c7-w8wwx               1/1     Running   0          5s
+pod/postgres-cluster-1                     1/1     Running   0          10m
+pod/postgres-cluster-2                     1/1     Running   0          10m
+pod/postgres-cluster-3                     1/1     Running   0          9m53s
+
+NAME                           TYPE        CLUSTER-IP       EXTERNAL-IP   PORT(S)           AGE
+service/cnpg-webhook-service   ClusterIP   10.104.218.212   <none>        443/TCP           20m
+service/ferretdb-service       NodePort    10.107.244.213   <none>        27017:30987/TCP   3m55s
+service/postgres-cluster-r     ClusterIP   10.107.141.185   <none>        5432/TCP          17m
+service/postgres-cluster-ro    ClusterIP   10.110.7.253     <none>        5432/TCP          17m
+service/postgres-cluster-rw    ClusterIP   10.98.97.107     <none>        5432/TCP          17m
+
+NAME                                  TYPE                       DATA   AGE
+secret/cnpg-ca-secret                 Opaque                     2      17m
+secret/cnpg-webhook-cert              kubernetes.io/tls          2      17m
+secret/ferretuser-secret              kubernetes.io/basic-auth   2      19m
+secret/postgres-cluster-ca            Opaque                     2      17m
+secret/postgres-cluster-replication   kubernetes.io/tls          2      17m
+secret/postgres-cluster-server        kubernetes.io/tls          2      17m
+secret/sh.helm.release.v1.cnpg.v1     helm.sh/release.v1         1      20m
+```
+
+### Connect to FerretDB via `mongosh`
+
+Create a temporary `mongosh` client to connect to the FerretDB instance.
+
+```sh
+kubectl -n cnpg run mongosh-client --image=rtsp/mongosh --rm -it -- bash
+```
+
+That will start up a `mongosh-client` within the namespace.
+
+```sh
+mongosh 'mongodb://ferretuser:<password>@ferretdb-service:27017/ferretdb?authMechanism=PLAIN'
```

Comment:
This wouldn't work we didn't create `ferretdb-service`, could it be `ferretdb`. Also above output from `kubectl get cluster,pod,svc,secret -n cnpg` seems outdate.